src/sentry/preprod/utils.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+import re
+from typing import NamedTuple
+
+
+class ParsedReleaseVersion(NamedTuple):
+    """Parsed components of a release version string."""
+
+    app_id: str
+    build_version: str
+
+
+def parse_release_version(release_version: str) -> ParsedReleaseVersion | None:
```

Comment:
Might be good to move this down 1 PR in the stack, but not a big deal

Reply:
Have you seen `from sentry_relay.processing import parse_release`?  It can be extended for your case if its not sufficient.

Reply:
I don't want to change that method because our use case would change how version is currently parsed. It'd go from:

```
Input:  'my-app-1.2.3@latest+11.1.2'
Output: {'package': 'my-app-1.2.3', 'version_raw': 'latest+11.1.2', 'version_parsed': None, 'build_hash': None, 'description': 'my-app-1.2.3@latest+11.1.2'}
```
to
```
Input:  'my-app-1.2.3@latest+11.1.2'
Output: {'package': 'my-app-1.2.3', 'version_raw': 'latest', 'version_parsed': None, build_number_raw:'11.1.2', 'build_hash': None, 'description': 'my-app-1.2.3@latest+11.1.2'}
```

so just going to leave this separate for now